airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -285,13 +288,13 @@ def _read_full_refresh(
                 if self._limit_reached(internal_config, total_records_counter):
                     return
 
-    def _checkpoint_state(self, stream, stream_state, connector_state):
+    @staticmethod
+    def _checkpoint_state(stream: Stream, stream_state, connector_state):
         try:
-            connector_state[stream.name] = stream.state
+            connector_state.update_state_for_stream(stream.name, stream.namespace, stream.state)
```

Comment:
I'm not sure this is a correct replacement. This line used to access `stream.state`, a property embedded inside the stream object itself containing its state. The function being called in the new code doesn't perform that access

Reply:
ignore me -- eyes weren't working that day :D

Reply:
discussed on slack. this should be working as expected using the `stream.state` override if it exists